launch/launch/__init__.py

```diff
@@ -75,8 +78,11 @@ def __init__(self, cmd, name, output_handler, exit_handler, env=None):
 
     def send_signal(self, signal):
         if self.transport:
+            self.task_state.signals_received.append(signal)
             self.transport.send_signal(signal)
 
     def terminate(self):
-        if self.transport:
+        if os.name != 'nt':
+            self.send_signal(signal.SIGINT)
```

Comment:
That will prevent the launcher to handle any further event on the concurrent futures: https://github.com/ros2/launch/blob/e3ec7bbfb180f8ede93a7ed85737d0def9c9c5bf/launch/launch/launcher.py#L190